src/core/net/socket.hpp

```diff
@@ -220,6 +220,22 @@ class MessageInfo : public otMessageInfo, public Clearable<MessageInfo>
      */
     void SetEcn(Ecn aEcn) { mEcn = aEcn; }
 
+    /**
+     * This method gets the DSCP value.
+     *
+     * @returns the DSCP value as represented in the IPv6 header.
+     *
+     */
+    uint8_t GetDscp(void) const { return mDscp; }
+
+    /**
+     * This method sets the DSCP value.
+     *
+     * @param[in] aDscp   The DSCP value.
```

Comment:
Should indicate here that only the lower 6 bits are used? Also the text could be more verbose e.g.

```
aDscp    The DSCP value to be represented in the IPv6 header.
```